modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -702,14 +748,32 @@ public boolean setTabSize(int spaces) {
      *                                                                         *
      **************************************************************************/
 
-    private int getLineIndex(float y) {
+    private int getLineIndex(float y, String text) {
         int index = 0;
         float bottom = 0;
+        /* Initializing textFound as true when text is null
+         * because when this function is called for TextFlow text parameter will be null */
+        boolean textFound = (text == null);
+
         int lineCount = getLineCount();
         while (index < lineCount) {
+            if (text != null) {
+                for (TextRun r : lines[index].runs) {
+                    if (r.getTextSpan() == null || r.getTextSpan().getText().equals(text)) {
```

Comment:
> (1) If you are looking for a particular match of text, what if there are 2 lines or spans that have the exact same text ? 

If 2 runs have the same text as it can happen in the case of emojis, line number will be calculated correctly. But if 2 text nodes have the exact same text and both are present in the same line, the character index calculation fails. I will check this case.

>(2) What is that the logic behind this line ? At least currently you only pass in text for a Text node. Can you break down what exactly it is doing ?

The line index calculation was failing for the Text node. In the case of Text node, even before the actual text line/run is reached, `bottom > y` was becoming true since it is relative to the text node. Hence added this condition of `r.getTextSpan().getText().equals(text)` to make sure that line index is calculated for the correct Text node.